indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalAppendAction.java

```diff
@@ -135,14 +135,16 @@ public SegmentPublishResult perform(Task task, TaskActionToolbox toolbox)
     if (startMetadata == null) {
       publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegments(
           segments,
```

Comment:
Do we need this branch at all? Can't we just simply call `commitAppendSegmentsAndMetadata()` with start and end metadata and remove `commitAppendSegments()`, if it's not anywhere else, which appears to be the case?